README.md

```diff
@@ -1,50 +1,84 @@
+  name: Generate API client using Hatch
   needs: <any prerequisite earlier job>
   runs-on: ubuntu-latest
   steps:
-    - uses: actions/checkout@v2 # <-- this one is needed for hatch to access source code
+    - uses: actions/checkout@v2
     - name: Run Hatch action
-      uses: birdiecare/hatch@v0.0.12 # <-- use current version
+      uses: birdiecare/hatch@v0.1.0 
       with:
-        path: <e.g. https://staging.myservice.com/docs.json>
-        name: <Name of your service, e.g. my-service> # the generated client will automatically be called my-service-client
-        token: <your GitHub Registry Token>
+        openapi_spec_url: https://my-service.io/docs.json
+        package_name: my-api-client
+        registry_namespace: @mycompany # the package will be published under @mycompany/my-api-client
+        registry_token: ${{ secrets.MY_REGISTRY_TOKEN }}
+        repository_url: ssh://git@github.com/my-organisation/my-service.git # the package will be published as part of this repo
 ```
 
-Running this Action will publish an NPM package called `my-service-client` as a package **inside the repository of the service running the Action**.
+### Exposing DTO classes using Hatch
+
+At [Birdie](https://birdie.care), we use [NestJS and Swagger](https://docs.nestjs.com/openapi/introduction) to annotate the DTO classes we use for our endpoints and RPC operations. For some of our use cases, it was helpful to have these full, Swagger-annotated DTO classes available as part of the generated client package, so that other services making use of this client have access to these annotations instead of just the interfaces provided by `openapi-generator`.
 
-### Exposing Swagger models
+To support this use case, the Hatch action can scan the `src` folder of your service's git repo for files matching the `*.public-models.ts` glob. If you wish to include Swagger-annotated (or other) DTO classes in the generated client package, then make sure to name them accordingly.
 
-If you want to create a proxy endpoint in `core-api`, you would like to import the class decorated with `@nestjs/swagger` decorators to avoid re-declaring the swagger docs.
+Once included, these DTOs can be imported from the generated client:
 
-In that case, the `birdiecare/hatch` action can scan your `src` folder of the git repo in search of all the files matching the `*.public-models.ts` glob. Other files will be ignored, so make sure you put the swagger classes in that file.
 
-Then in the `core-api` you can import the file from `package/models/filename` nested import, e.g.:
 ```ts
-import { IdentifierMapping } from '@birdiecare/rostering-integrations-client/models/identifier-mapping.public-models';
+import { MyDTO } from '@mycompany/my-service-client/models/my-models.public-models';
 ```
+⚠️  Note that public model files currently do not support any other imports than `@nestjs/swagger` and `class-validator`. If you import anything else, they will fail to compile.
```

Comment:
```suggestion
⚠️  Note that public model files currently do not support any other imports than `@nestjs/swagger` and `class-validator`. If you import anything else, they will fail to compile. These two dependencies are peer-dependencies of the generated client package and should be installed in your consuming project if you use the public models feature.
```
Do we make them peer dependencies of the client?

Reply:
Yes :-)